src/rules/no-missing-link-fragments.js

```diff
@@ -57,81 +78,101 @@ export default {
 		],
 
 		messages: {
-			missingFragment:
-				"Link fragment '#{{fragment}}' does not exist in the document.",
+			invalidFragment:
+				"Link fragment '{{fragment}}' does not reference a valid heading or anchor.",
```

Comment:
I think this message is confusing.

```suggestion
				"Link fragment '{{fragment}}' does not reference a heading or anchor in this document.",
```